satpy/tests/writer_tests/test_cf.py

```diff
@@ -44,28 +59,79 @@ def test_init(self):
     def test_save_array(self):
         from satpy import Scene
         import xarray as xr
-        import tempfile
         scn = Scene()
         start_time = datetime(2018, 5, 30, 10, 0)
         end_time = datetime(2018, 5, 30, 10, 15)
         scn['test-array'] = xr.DataArray([1, 2, 3],
                                          attrs=dict(start_time=start_time,
                                                     end_time=end_time,
                                                     prerequisites=[DatasetID('hej')]))
-        try:
-            handle, filename = tempfile.mkstemp()
+        with TempFile() as (handle, filename):
             os.close(handle)
-            scn.save_datasets(filename=filename, writer='cf')
+            scn.save_datasets(filename=filename, writer='cf', engine='h5netcdf')
```

Comment:
The existing tests use `h5netcdf` for reading, but some files written by the `netcdf4` engine (the new default engine atm) are not readable by `h5netcdf` (errors like https://github.com/h5py/h5py/issues/1180). That is why I changed the engine to `h5netcdf` in these calls. If we decided to keep `netcdf4` as default engine I would update these tests to use `netcdf4` for reading.